source/hardware_init.c

```diff
@@ -0,0 +1,89 @@
+/*
```

Comment:
You should change the name of this file - maybe retain the old name? You copied it from LWIP hardware_init.c, and it would clash in some of the build systems if there were two files with the same leafname.

(Yes, I know this is ridiculous).